art/attacks/evasion/auto_conjugate_gradient.py

```diff
@@ -0,0 +1,714 @@
+                            condition = np.logical_or(condition_1, condition_2)
+
+                            # halve the stepsize if the condition is satisfied
+                            eta[condition] /= 2
+                            # move to the best point
+                            x_max_tmp = self.x_max[condition].copy()
+                            x_k[condition] = x_max_tmp.copy()
+                            gradk_1_tmp = gradk_1_best[condition].copy()
+                            gradk_1[condition] = gradk_1_tmp.copy()
+                            sk_1_tmp = sk_1_best[condition].copy()
+                            sk_1[condition] = sk_1_tmp.copy()
+
+                            self.count_condition_1[:] = 0
+                            self.eta_w_j_m_1 = eta.copy()
+                            self.f_max_w_j_m_1 = self.f_max.copy()
+
+                y_pred_adv_k = self.estimator.predict(x_k)
+                if self.targeted:
+                    sample_is_not_robust_k = np.invert(np.argmax(y_pred_adv_k, axis=1) != np.argmax(y_batch, axis=1))
+                elif not self.targeted:
+                    sample_is_not_robust_k = np.invert(np.argmax(y_pred_adv_k, axis=1) == np.argmax(y_batch, axis=1))
+
+                x_robust[batch_index_1:batch_index_2][sample_is_not_robust_k] = x_k[sample_is_not_robust_k]
+
+            x_adv[sample_is_robust] = x_robust
+
+        return x_adv
+
+    def _check_params(self) -> None:
+        if self.norm not in [1, 2, np.inf, "inf"]:
+            raise ValueError('The argument norm has to be either 1, 2, np.inf, or "inf".')
+
+        if not isinstance(self.eps, (int, float)) or self.eps <= 0.0:
+            raise ValueError("The argument eps has to be either of type int or float and larger than zero.")
+
+        if not isinstance(self.eps_step, (int, float)) or self.eps_step <= 0.0:
+            raise ValueError("The argument eps_step has to be either of type int or float and larger than zero.")
+
+        if not isinstance(self.max_iter, int) or self.max_iter <= 0:
+            raise ValueError("The argument max_iter has to be of type int and larger than zero.")
+
+        if not isinstance(self.targeted, bool):
+            raise ValueError("The argument targeted has to be of bool.")
+
+        if not isinstance(self.nb_random_init, int) or self.nb_random_init <= 0:
+            raise ValueError("The argument nb_random_init has to be of type int and larger than zero.")
+
+        if not isinstance(self.batch_size, int) or self.batch_size <= 0:
+            raise ValueError("The argument batch_size has to be of type int and larger than zero.")
+
+        # if self.loss_type not in self._predefined_losses:
+        #     raise ValueError("The argument loss_type has to be either {}.".format(self._predefined_losses))
+
+        if not isinstance(self.verbose, bool):
+            raise ValueError("The argument `verbose` has to be of type bool.")
+
+
+def getBeta(gradk, gradk_1, sk_1):
```

Comment:
snake_case naming style:
```suggestion
def get_beta(gradk, gradk_1, sk_1):
```